broadcaster/assets/js/home.js

```diff
@@ -83,10 +83,13 @@ async function connectInput(id) {
     videoPlayerWrapper.appendChild(videoPlayer);
     inputData.videoPlayer = videoPlayer;
     updateVideoGrid();
-    whepClient.changeLayer(defaultLayer);
     statusMessage.classList.add('hidden');
   };
 
+  whepClient.onconnected = () => {
+    whepClient.changeLayer(defaultLayer);
+  };
+
```

Comment:
This helps not to send the layer change request too early (but doesn't solve our problem with missing keyframes, unfortunately)

Reply:
Btw. the server should be prepared for incorrect request anyway :D